src/java/org/apache/cassandra/dht/RangeStreamer.java

```diff
@@ -353,9 +361,27 @@ public void addRanges(String keyspaceName, ReplicaCollection<?> replicas)
      */
     private boolean useStrictSourcesForRanges(AbstractReplicationStrategy strat)
     {
-        return useStrictConsistency
-                && tokens != null
-                && metadata.getSizeOfAllEndpoints() != strat.getReplicationFactor().allReplicas;
+        boolean res = useStrictConsistency && tokens != null;
+        
+        if (res)
+        {
+            int nodes = 0;
+
+            if (strat instanceof NetworkTopologyStrategy)
```

Comment:
Sorry, I am not sure what the conditions should be, just to make sure:
DC1:3, DC2:3 with DC1 having 2 and DC2 having 6 nodes => res = True?

Also the condition before change was `!=` and now it is `>`?

Reply:
I am not following your explanation sorry.

In any case notice we're comparing the number of _replicating nodes_ (not nodes, but replicating nodes only) to the RF. So if you have 6 nodes but only 3 are _replicas_ you should behave as if there were only 3 nodes towards the RF, not 6 bc one dc is _not_ replicating i.e. Does it help?

Yep method doc also points towards `>`  which makes sense to me. Idk why it was `!=`

Reply:
ok, so maybe I was not precise, feel free to ignore:
if we have: NTS witrh RF = {DC1:3, DC2: 3} and we have cluster with N(DC1)=2, N(DC2)=6 then
- nodes will be 8?
- so res will be True?

What is bit misleading in your explanation:
> not 6 bc one dc is not replicating

Reply:
Ok icwym now. You were talking DC:RF and DC:N on each side. Yes, nodes would be 8. And on DC1 you'd get a warning bd RF=3 but N=2.